saleor/graphql/decorators.py

```diff
@@ -23,15 +23,30 @@ def wrapper(context, *args, **kwargs):
 
 def permission_required(perm):
     def check_perms(context):
-        if isinstance(perm, six.string_types):
-            perms = (perm,)
-        else:
-            perms = perm
-        if context.user.has_perms(perms):
-            return True
-        service_account = getattr(context, "service_account", None)
-        if service_account and service_account.has_perms(perms):
-            return True
+        return _permission_required(perm, context)
+
+    return account_passes_test(check_perms)
+
+
+def one_of_permissions_required(perms):
+    def check_perms(context):
+        for perm in perms:
+            has_perm = _permission_required(perm, context)
+            if has_perm:
+                return True
         return False
 
     return account_passes_test(check_perms)
+
+
+def _permission_required(perm, context):
+    if isinstance(perm, six.string_types):
+        perms = (perm,)
+    else:
+        perms = perm
+    if context.user.has_perms(perms):
+        return True
+    service_account = getattr(context, "service_account", None)
+    if service_account and service_account.has_perms(perms):
+        return True
+    return False
```

Comment:
```python
    user = context.user or getattr(context, "service_account", None)

    if user and has_perm: ...
```

Reply:
We use to check it in this order because `context.user` have `AnonymousUser` for a service account. To make your solution to work, we need to add a condition for checking `AnonymousUser`. I think the current solution is better.

Reply:
True, true, true. Forgot about `user` not being `None` in Django 👀